src/components/settingsModal/settingsModal.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useState } from 'react'
+import './styles.css'
+// import { invoke } from "@tauri-apps/api/tauri"
+
+interface ModalProps {
+  show: boolean;
+  onClose: () => void;
+}
+
+const SettingsModal: React.FC<ModalProps> = ({ show, onClose }) => {
+  if (!show) return null;
+
+  // TODO: get the users current permisions json
+  const [toggles, setToggles] = useState({
+    Location: false,
+    Clipboard: false,
+    Screenshot: false,
+  });
+
+  const handleToggle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setToggles({ ...toggles, [event.target.name]: event.target.checked });
+  };
+
+  useEffect(() => {
+    // TODO: update the permissions.json
+    console.log(toggles)
+  }, [toggles])
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      const modal = document.querySelector('.large-modal');
+      if (modal && !modal.contains(event.target as Node)) {
+        onClose();
+      }
+    };
+
+    if (show) {
+      document.addEventListener('mousedown', handleClickOutside);
+    }
+
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
```

Comment:
I think im confused on what this section is doing. Will the listener always be removed when this useEffect is ran? should it only be removed is we aren't showing the modal?

Reply:
it removes the event listener when the settings modal is unmounted. so that every time we show the modal, and then close it, we make sure to remove the previous event listener when we open the modal again